pkg/cluster/k8sres.go

```diff
@@ -610,6 +610,36 @@ func generatePodAntiAffinity(podAffinityTerm v1.PodAffinityTerm, preferredDuring
 	return podAntiAffinity
 }
 
+func generateTopologySpreadConstraints(labels labels.Set, topologySpreadConstraintObjs []v1.TopologySpreadConstraint) []v1.TopologySpreadConstraint {
```

Comment:
would like to see a unit test for this function :)

Reply:
I added a unit test for this function.